instruments/gentec_eo/blu.py

```diff
@@ -0,0 +1,647 @@
+        max10 = "26"
+        max30 = "27"
+        max100 = "28"
+        max300 = "29"
+        max1kilo = "30"
+        max3kilo = "31"
+        max10kilo = "32"
+        max30kilo = "33"
+        max100kilo = "34"
+        max300kilo = "35"
+        max1Mega = "36"
+        max3Mega = "37"
+        max10Mega = "38"
+        max30Mega = "39"
+        max100Mega = "40"
+        max300Mega = "41"
+
+    # PROPERTIES #
+
+    @property
+    def anticipation(self):
+        """Get / Set anticipation.
+
+        This command is used to enable or disable the anticipation
+        processing when the device is reading from a wattmeter. The
+        anticipation is a software-based acceleration algorithm that
+        provides faster readings using the detector’s calibration.
+
+        :return: Is anticipation enabled or not.
+        :rtype: bool
+
+        Example:
+            >>> import instruments as ik
+            >>> inst = ik.gentec_eo.Blu.open_serial('/dev/ttyACM0')
+            >>> inst.anticipation
+            True
+            >>> inst.anticipation = False
+        """
+        return self._value_query("*GAN", tp=int) == 1
+
+    @anticipation.setter
+    def anticipation(self, newval):
+        sendval = 1 if newval else 0
+        self.sendcmd("*ANT{}".format(sendval))
+
+    @property
+    def auto_scale(self):
+        """Get / Set auto scale on the device.
+
+        :return: Status of auto scale enabled feature.
+        :rtype: bool
+
+        :raises ValueError: The command was not acknowledged by the
+            device.
+
+        Example:
+            >>> import instruments as ik
+            >>> inst = ik.gentec_eo.Blu.open_serial('/dev/ttyACM0')
+            >>> inst.auto_scale
+            True
+            >>> inst.auto_scale = False
+        """
+        resp = self._value_query("*GAS", tp=int)
+        return resp == 1
+
+    @auto_scale.setter
+    def auto_scale(self, newval):
+        sendval = 1 if newval else 0
+        self.sendcmd("*SAS{}".format(sendval))
+
+    @property
+    def available_scales(self):
+        """Get available scales from connected device.
+
+        :return: Scales currently available on device.
+        :rtype: :class:`Blu.Scale`
+
+        Example:
+            >>> import instruments as ik
+            >>> inst = ik.gentec_eo.Blu.open_serial('/dev/ttyACM0')
+            >>> inst.available_scales
+            [<Scale.max100milli: '22'>, <Scale.max300milli: '23'>,
+            <Scale.max1: '24'>, <Scale.max3: '25'>, <Scale.max10: '26'>,
+            <Scale.max30: '27'>, <Scale.max100: '28'>]
+        """
+        # set no terminator and a 1 second timeout
+        _terminator = self.terminator
+        self.terminator = ""
+        _timeout = self.timeout
+        self.timeout = u.Quantity(1, u.s)
+
+        # get the response
+        resp = self._no_ack_query("*DVS").split('\r\n')
+
+        # set back terminator and 3 second timeout
+        self.terminator = _terminator
```

Comment:
Potential issue here: if the above line (189) errors out, then our temporary variable changes here are never reset

Reply:
Great catch! Taking care of this now with a try, finally block to reset the temporary variables even if the read errors out. Also added the appropriate test to ensure this is actually working.